server/src/main/java/com/google/musicanalysis/site/YoutubeServlet.java

```diff
@@ -15,39 +17,106 @@
+     *  gets json string of youtube response
+     * @param apiKey youtube api key
+     * @param accessToken youtube access token from login
+     * @return JSON string of youtube response of liked video topics
+     * @throws ServletException
+     * @throws IOException
+     */
+    protected String getYoutubeRes(String apiKey, String accessToken) 
         throws ServletException, IOException {
-        // retrieve Youtube API key and access token 
-        String API_KEY = Secrets.getSecretString("YOUTUBE_API_KEY");
-        HttpSession session = req.getSession();
-        var accessToken = session.getAttribute("oauth-access-token-youtube");
-        if (accessToken == null) {
-            res.setStatus(401);
-            return;
-        }
-    
         // make http request to youtube API
-        var youtubeParam = new URLEncodedBuilder()
+        URLEncodedBuilder youtubeParam = new URLEncodedBuilder()
             .add("part", "topicDetails")
             .add("myRating", "like")
-            .add("key", API_KEY);
+            .add("key", apiKey);
         URI youtubeUri = URI.create("https://www.googleapis.com/youtube/v3/videos?" + youtubeParam.build());
 
         var httpClient = HttpClient.newHttpClient();
-        var youtubeReq = HttpRequest.newBuilder(youtubeUri)
-                .header("Authorization", "Bearer " + accessToken.toString())
+        HttpRequest youtubeReq = HttpRequest.newBuilder(youtubeUri)
+                .header("Authorization", "Bearer " + accessToken)
                 .header("Accept", "application/json")
                 .GET()
                 .build();
 
-        // get response to Youtube API 
+        // get response from Youtube API 
         var youtubeRes = httpClient.sendAsync(youtubeReq, BodyHandlers.ofString()).join();
-        var youtubeResBody = youtubeRes.body();
-        res.getWriter().write(youtubeResBody);
+        return youtubeRes.body();
+    }
+
+    /**
+     * checks whether topic is categorized as music
+     * by checking if the last word is "music" or "Music"
+     * @param topic identifies youtube video category e.g. Knowledge or Pop music
+     * @return whether topic is categorized as music
+     */
+    protected Boolean isMusic(String topic) {
+        String lastWord = topic.substring(topic.lastIndexOf(" ") + 1);
+        return lastWord.equalsIgnoreCase("music");
+    }
+
+    /**
+     * parses through youtube liked videos json string,
+     * updates hash map to contain frequency count of each music genre
+     * @param youtubeResBody json response of youtube liked videos
+     * @param genreCount hash map of frequency count of each music genre
+     */
+    protected void updateMusicCount(String youtubeResBody, HashMap<String, Integer> genreCount) {
+        JsonObject jObject = JsonParser.parseString(youtubeResBody).getAsJsonObject();
+        JsonArray videos = jObject.getAsJsonArray("items");
+
+        // iterates through each liked video
```

Comment:
You don't need comments like these that just explain the mechanics of the code (in this case that the for loop is iterating over a list of videos). Reserve implementation comments for places where the intention of the code isn't immediately obvious.